jerry-core/ecma/operations/ecma-jobqueue.h

```diff
@@ -0,0 +1,69 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ECMA_JOB_QUEUE_H
+#define ECMA_JOB_QUEUE_H
+
+#ifndef CONFIG_DISABLE_ES2015_PROMISE_BUILTIN
+
+/** \addtogroup ecma ECMA
+ * @{
+ *
+ * \addtogroup ecmajobqueue ECMA Job Queue related routines
+ * @{
+ */
+typedef enum
+{
+  ECMA_JOB_TYPE_PROMISE_REACTION_JOB, /**< PromiseReactionJob ES2015 25.4.2.1*/
+  ECMA_JOB_TYPE_PROMISE_THENABLE_JOB, /**< PromiseResolveThenableJob ES2015 25.4.2.2*/
+  ECMA_JOB_TYPE__COUNT /**< count of the job types */
+} ecma_job_type_t;
+
+/**
+ * Description of general ecma job.
+ * Usually, it is used as the header of a specific job
+ */
+typedef struct
+{
+  ecma_job_type_t type; /**< specific type of the job */
```

Comment:
enums are compiler specific, sometimes int, sometimes unsigned char. I would prefer to use an uint8_t here.